rotkehlchen/premium/premium.py

```diff
@@ -234,57 +276,90 @@ def authenticate_device(self) -> None:
+            )
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Failed to check device registration due to: {e}') from e
 
-        device_id = machineid.hashed_id(self.username)
+        if response.status_code == HTTPStatus.FORBIDDEN:
+            raise PremiumAuthenticationError('Premium credentials not valid')
+
+        if response.status_code == HTTPStatus.OK:
+            # Device is already registered
+            return None
+
+        if response.status_code == HTTPStatus.NOT_FOUND:
+            # Device is not registered, try to register it
+            return self._register_new_device(device_id)
 
-        for device in device_data['devices']:
-            device = cast('dict[str, str]', device)
-            if (remote_id := device.get('device_identifier')) == device_id:
-                break
-            if remote_id is None:
-                log.error(f'Remote device {device} has no identifier in the server response')
-        else:  # device not found
-            if num_devices < devices_limit:
-                # try to register the device
-                self._register_new_device(device_id)
-            else:
-                # user has to edit his devices
-                raise PremiumAuthenticationError(
-                    f'The limit of {devices_limit} devices has been reached',
-                )
-
-    def _register_new_device(self, device_id: str) -> dict:
+        return check_response_status_code(
+            response=response,
+            status_codes=[HTTPStatus.OK],
+        )
+
+    def _register_new_device(self, device_id: str) -> None:
         """
         Register a new device at the rotki server using the provided id.
         May raise:
         - RemoteError
         - PremiumAuthenticationError: if the queried API returns a 401 error
         """
         log.debug(f'Registering new device {device_id}')
-        method = 'devices'
-        device_name = platform.system()
-        data = self.sign(
-            method=method,
-            device_identifier=device_id,
-            device_name=device_name,
-        )
-
         try:
             response = self.session.put(
-                url=f'{self.rotki_api}{method}',
-                data=data,
+                url=f'{self.rotki_nest}{(method := "devices")}',
+                json=self.sign(
+                    method=method,
+                    device_identifier=device_id,
+                    device_name=str(uuid4()),  # can be edited later.
+                    platform=platform.system(),
+                ),
             )
         except requests.exceptions.RequestException as e:
             raise RemoteError(f'Failed to register device due to: {e}') from e
 
-        return _process_dict_response(response)
+        if response.status_code == HTTPStatus.CREATED:
+            return None
+
+        if (
+            response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR and
```

Comment:
That does not make sense. A device already exists should be a specific code. And it should definitely absolutely not be an INTERNAL_SERVER_ERROR. It needs to be a specific code, with a specific status to check. Not this.